src/containers/AssetDrawer.tsx

```diff
@@ -52,9 +36,36 @@ const HeaderWithButton = styled.div`
   justify-content: space-between;
 `;
 
-class AssetDrawer extends React.Component {
-  state = {
-    showAddTimeseries: false,
+type OrigProps = {
+  asset: ExtendedAsset;
+  width: number;
+};
+
+type Props = {
+  asset: ExtendedAsset;
+  doFetchTimeseries: typeof fetchTimeseries;
+  doFetchEvents: typeof fetchEvents;
+  doRemoveAssetFromTimeseries: typeof removeAssetFromTimeseries;
+  doRemoveTypeFromAsset: typeof removeTypeFromAsset;
+  timeseries: TimeseriesState;
+  events: EventsAndTypes;
+  threed: ThreeDState;
+  types: TypesState;
+  width: number;
+};
+
+type State = {
+  showAddTimeseries?: boolean;
+  asset?: ExtendedAsset;
+  showAddTypes?: boolean;
```

Comment:
Why are these undefined by default?

Reply:
hmm `showAddTypes` can also be a boolean i believe.